src/mmw/js/src/modeling/models.js

```diff
@@ -130,12 +163,32 @@ var ScenarioModel = Backbone.Model.extend({
         name: '',
         is_current_conditions: false,
         modifications: null,      // ModificationsCollection
-        active: false
+        active: false,
+        uid: null
     },
 
     initialize: function(attrs, options) {
         Backbone.Model.prototype.initialize.apply(this, arguments);
+        this.set('uid', App.user.get('uid'));
         this.set('modifications', ModificationsCollection.create(attrs.modifications));
+        this.on({
+            'change:project': this.attemptSave,
+            'change:name': this.attemptSave
+        }, this);
+
+        this.get('modifications').on({
+            'add': this.attemptSave,
+            'remove': this.attemptSave
+        }, this);
+    },
+
+    attemptSave: function() {
```

Comment:
Consider letting the server fail if a non-owner tries to save. We shouldn't display save buttons for non-owners.